### ISSUE TITLE
different ports shown on same page

### ISSUE THREAD

**Issue by rajdeepbharati** [macports/buildbot-macports-custom-views#26]
Builds for py-tensorflow and py-tensorflow_estimator are displayed on the same page.
http://ec2-3-17-139-39.us-east-2.compute.amazonaws.com/#/ports?id=py-tensorflow

**Comment by ryandesign** [comment]
At least one problem is here:

https://github.com/macports/buildbot-macports-custom-views/blob/master/src/components/PortsList.vue#L92-L95

```
                if (builder.name.includes('_')) {
                  return (
                    builder.name
                      .slice(0, builder.name.indexOf('_'))
                      .toLowerCase()
                      .indexOf(this.search.toLowerCase()) > -1
```

This assumes that a builder name is *portname*_*platform* and that *portname* does not contain underscores. The first assumption isn't correct for my installation because I changed the format of the builder names, and the second assumption isn't correct for ports like py-tensorflow_estimator whose names do contain underscores.

Can it access properties of the build? We set the `portname` build property to the name of the port.

**Comment by rajgoesout** [comment]
> This assumes that a builder name is portname_platform and that portname does not contain underscores. The first assumption isn't correct for my installation because I changed the format of the builder names, and the second assumption isn't correct for ports like py-tensorflow_estimator whose names do contain underscores.

You're right. I had assumed some weird naming conventions.

> Can it access properties of the build? We set the portname build property to the name of the port.

I'm not sure whether that could be done efficiently in the frontend (I'll check and let you know), but we could assign a tag which would basically be the portname. I think we are already doing that: https://github.com/macports-gsoc/macports-buildbot/blob/master/master.cfg#L135

**Comment by ryandesign** [comment]
Yes there is a tag for the port name but there are also other tags (`_virtual_` and either the macOS version or the platform for non-Darwin workers). Can you differentiate between the tags to find the one that is the port name?

**Comment by rajgoesout** [comment]
We can assign tags in a particular order (in the buildmaster) such that the first tag (in the list of tags) is always the portname. Then it would be possible to differentiate.

**Comment by ryandesign** [comment]
If it preserves the order, then sure.
